### PR TITLE
JsDoc cleanup

### LINKED ISSUE #617: Consistency codebase refactor
Let's do some refactoring to have a more consistent codebase:

- [x] It seems that we are more inclined to use types without `T` prefix. Let's unify for consistency. https://github.com/scaffold-eth/scaffold-eth-2/pull/627
- [x] `0x${string}` to viem's `Address`? [WIP @sverps ]
- [x] TS: interfaces -> types https://github.com/scaffold-eth/scaffold-eth-2/pull/627
- [ ] JSDoc tweaks: see https://github.com/scaffold-eth/scaffold-eth-2/pull/627#issuecomment-1828163974

Check TS guide: https://github.com/scaffold-eth/scaffold-eth-2/issues/232

What else? cc @technophile-04 @rin-st @sverps

### PR DESCRIPTION
**Pull request by technophile-04**
## Description

4th TODO from https://github.com/scaffold-eth/scaffold-eth-2/issues/617#issue-2003940907, also fixes #617

### CODE REVIEW

File: packages/nextjs/hooks/scaffold-eth/useDeployedContractInfo.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { usePublicClient } from "wagmi";
 import { Contract, ContractCodeStatus, ContractName, contracts } from "~~/utils/scaffold-eth/contract";
 
 /**
- * Gets the matching contract info from the contracts file generated by `yarn deploy`
+ * Gets the matching contract info from the contracts present in deployedContracts.ts and externalContracts.ts
  * @param contractName - name of deployed contract
```
Comment (rin-st) [inline packages/nextjs/hooks/scaffold-eth/useDeployedContractInfo.ts:9]: I think it's better to also delete this param

File: packages/nextjs/hooks/scaffold-eth/useOutsideClick.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import React, { useEffect } from "react";
 
 /**
- * Check if a click was made outside the passed ref
+ * Checks if a click was made outside of passed ref
```
Comment (rin-st) [inline packages/nextjs/hooks/scaffold-eth/useOutsideClick.ts:4]: It's not only check, main thing here is handling click outside (callback).

So I think it should be something like `Handles clicks outside of passed ref`

File: packages/nextjs/hooks/scaffold-eth/useScaffoldContract.ts [hunk 3]
```diff
@@ -5,10 +5,10 @@ import { useDeployedContractInfo } from "~~/hooks/scaffold-eth";
 import { Contract, ContractName } from "~~/utils/scaffold-eth/contract";
 
 /**
- * Gets a deployed contract by contract name and returns a contract instance
- * @param config - The config settings
- * @param config.contractName - Deployed contract name
- * @param config.walletClient - An viem wallet client instance (optional)
+ * Gets an instance of contract for the given contract name. Optional walletClient can be passed for doing write transactions.
+ * @param config - The config settings for the hook
```
Comment (rin-st) [inline packages/nextjs/hooks/scaffold-eth/useScaffoldContract.ts:9]: there's no `config` param in hook 🤷‍♂️

File: packages/nextjs/hooks/scaffold-eth/useScaffoldContractWrite.ts [hunk 4]
```diff
@@ -10,10 +10,9 @@ import { ContractAbi, ContractName, UseScaffoldWriteConfig } from "~~/utils/scaf
...
 /**
- * Wrapper for wagmi's useContractWrite hook (with config prepared by usePrepareContractWrite hook)
- * which automatically loads (by name) the contract ABI and address from the deployed contracts
+ * Wrapper around wagmi's useContractWrite hook which automatically loads (by name) the contract ABI and address from the contracts present in deployedContracts.ts & externalContracts.ts
  * @param config - The config settings, including extra wagmi configuration
- * @param config.contractName - deployed contract name
+ * @param config.contractName - contract name
  * @param config.functionName - name of the function to be called
  * @param config.args - arguments for the function
  * @param config.value - value in ETH that will be sent with transaction
```
Comment (rin-st) [inline packages/nextjs/hooks/scaffold-eth/useScaffoldContractWrite.ts:18]: no information about onBlockConfirmation and blockConfirmations,

**Comment by rin-st** [comment]
Gj Shiv, thanks! 🙌  Left some comments

File: packages/nextjs/hooks/scaffold-eth/useDeployedContractInfo.ts (hunk 1 above)
Comment (technophile-04) [reply packages/nextjs/hooks/scaffold-eth/useDeployedContractInfo.ts]: yup done 🙌 at https://github.com/scaffold-eth/scaffold-eth-2/pull/665/commits/beb5659bc39a8d5718deffa73121d09c7ae4f472

File: packages/nextjs/hooks/scaffold-eth/useOutsideClick.ts (hunk 2 above)
Comment (technophile-04) [reply packages/nextjs/hooks/scaffold-eth/useOutsideClick.ts]: Oh yes, thanks! fixed at [4b737b5](https://github.com/scaffold-eth/scaffold-eth-2/pull/665/commits/4b737b594e65216499a5cc4158ac3d72ad5cc0db)

File: packages/nextjs/hooks/scaffold-eth/useScaffoldContract.ts (hunk 3 above)
Comment (technophile-04) [reply packages/nextjs/hooks/scaffold-eth/useScaffoldContract.ts:9]: Ohh lol yeah !! , even I was wondering initially but it seems a standard for documenting object properties : 

Screenshot 2023-12-18 at 11 39 29 AM
https://jsdoc.app/tags-param

File: packages/nextjs/hooks/scaffold-eth/useScaffoldContractWrite.ts (hunk 4 above)
Comment (technophile-04) [reply packages/nextjs/hooks/scaffold-eth/useScaffoldContractWrite.ts:18]: Yup updated at [b9596e5](https://github.com/scaffold-eth/scaffold-eth-2/pull/665/commits/b9596e525b317f67e5fe780228eb4a042374b235) thanks 🙌

**Review by rin-st — APPROVED** [review]
lgtm!

**Comment by carletex** [comment]
Thanks @technophile-04

Love clean-up PRs :D

**Pull request merged**